### ISSUE TITLE
feat: additional field to workflow description

### ISSUE THREAD

**Issue by pkarw** [callstackincubator/fabrice-ai#99]
Right now, the entire workflow and what has been done so far are shared between agents. The idea is to separate a part of the workflow context, which is shared with agents. Otherwise, the agents use this workflow information to plan their own steps, which sometimes leads to performing tasks that were already executed or duplicating work.

Example:

```ts
const researchTripWorkflow = workflow({
  members: [personalizedActivityPlanner, restaurantScout, landmarkScout, itineraryCompiler],
  description: `
    Research and find cool things to do in Wrocław, Poland.

    Focus:
      - activities and events that match the traveler's interests and age group.
      - highly-rated restaurants and dining experiences.
      - landmarks with historic context.
      - picturesque and entertaining locations.

    Traveler's information:
      - Origin: New York, USA
      - Destination: Wrocław, Poland
      - Age of the traveler: 30
      - Hotel location: Main Square, Wrocław
      - Flight information: Flight AA123, arriving on 2023-12-15
      - How long is the trip: 7 days
  `,
  output: `
    Comprehensive day-by-day itinerary for the trip to Wrocław, Poland.
    Ensure the itinerary integrates flights, hotel information, and all planned activities and dining experiences.
  `,
  snapshot: logger,
})
```

after refactor:

```ts
const researchTripWorkflow = workflow({
  members: [personalizedActivityPlanner, restaurantScout, landmarkScout, itineraryCompiler],
  description: `
    Research and find cool things to do in Wrocław, Poland.

    Focus:
      - activities and events that match the traveler's interests and age group.
      - highly-rated restaurants and dining experiences.
      - landmarks with historic context.
      - picturesque and entertaining locations.
  `,
  output: `
    Comprehensive day-by-day itinerary for the trip to Wrocław, Poland.
    Ensure the itinerary integrates flights, hotel information, and all planned activities and dining experiences.
  `,
  knowledge: `
    Traveler's information:
      - Origin: New York, USA
      - Destination: Wrocław, Poland
      - Age of the traveler: 30
      - Hotel location: Main Square, Wrocław
      - Flight information: Flight AA123, arriving on 2023-12-15
      - How long is the trip: 7 days
      `
  snapshot: logger,
})
```

not sure if `knowledge` is the best name however `context` could be a little bit confusing as it appears in other places

**Comment by pkarw** [comment]
... and only `knowledge` is being shared with each agent

**Comment by pkarw** [comment]
Somehow related to #95

**Comment by grabbou** [comment]
I think that’s good name to start with!

**Comment by pkarw** [comment]
Eventually closed with #111
